gazelle/internal/spdump/target_dependency.go

```diff
@@ -45,7 +45,7 @@ func (pr *ProductReference) UnmarshalJSON(b []byte) error {
 		return err
 	}
 	if pr.DependencyName, err = jsonutils.StringAtIndex(raw, 1); err != nil {
-		return err
+		pr.DependencyName = pr.ProductName
```

Comment:
I went looking for a reference in the SPM code for how they interpret the single name.  I found [this](https://github.com/apple/swift-package-manager/blob/main/Sources/PackageModel/Target.swift#L59-L62).

>// this is hit only if this product is referenced `.byName(name)`
// which assumes the name of this product, its package, and its target
// all have the same name

So, I think that your change looks good to me.

Reply:
wow, nice to know that.